go-controller/pkg/metrics/metrics.go

```diff
@@ -144,6 +172,177 @@ func coverageShowMetricsUpdater(component string) {
+	default:
+		return nil, fmt.Errorf("unknown component %s for stopwatch/show", component)
+	}
+
+	if err != nil {
+		return nil, fmt.Errorf("failed to get stopwatch/show output for %s (stderr: %s): %w", component, stderr, err)
+	}
+
+	parsedOutput := parseStopwatchShowOutput(stdout)
+	return parsedOutput, err //need to return err, as it could be set in the defer
+}
+
+// parseStopwatchShowOutput returns the number of total samples for each poll loop
+func parseStopwatchShowOutput(output string) map[string]stopwatchStatistics {
+	result := make(map[string]stopwatchStatistics)
+
+	reStatisticsBlock := regexp.MustCompile(`(?m)^Statistics for '(?P<name>.+)'$(?:\n  .*)+`)
+	reTotalSamples := regexp.MustCompile(`(?m)^  Total samples: (?P<totalSamples>\d*)$`)
+	reMaximum := regexp.MustCompile(`(?m)^  Maximum: (?P<max>\d*) msec$`)
+	reMinumum := regexp.MustCompile(`(?m)^  Minimum: (?P<min>\d*) msec$`)
+	rePercentile95th := regexp.MustCompile(`(?m)^  95th percentile: (?P<percentile95th>\d*\.?\d*) msec$`)
+	reShortTermAvg := regexp.MustCompile(`(?m)^  Short term average: (?P<shortTermAvg>\d*\.?\d*) msec$`)
+	reLongTermAvg := regexp.MustCompile(`(?m)^  Long term average: (?P<longTermAvg>\d*\.?\d*) msec$`)
+
+	nameIndex := reStatisticsBlock.SubexpIndex("name")
+	totalSamplesIndex := reTotalSamples.SubexpIndex("totalSamples")
+	maxIndex := reMaximum.SubexpIndex("max")
+	minIndex := reMinumum.SubexpIndex("min")
+	percentile95thIndex := rePercentile95th.SubexpIndex("percentile95th")
+	shortTermAvgIndex := reShortTermAvg.SubexpIndex("shortTermAvg")
+	longTermAvgIndex := reLongTermAvg.SubexpIndex("longTermAvg")
+
+	for _, match := range reStatisticsBlock.FindAllStringSubmatch(output, -1) {
+		sws := stopwatchStatistics{}
+
+		if totalSamplesMatch := reTotalSamples.FindStringSubmatch(match[0]); totalSamplesMatch != nil {
+			sws.totalSamples = totalSamplesMatch[totalSamplesIndex]
+		}
+		if minMatch := reMinumum.FindStringSubmatch(match[0]); minMatch != nil {
+			sws.min = minMatch[minIndex]
+		}
+		if maxMatch := reMaximum.FindStringSubmatch(match[0]); maxMatch != nil {
+			sws.max = maxMatch[maxIndex]
+		}
+		if percentile95thMatch := rePercentile95th.FindStringSubmatch(match[0]); percentile95thMatch != nil {
+			sws.percentile95th = percentile95thMatch[percentile95thIndex]
+		}
+		if shortTermAvgMatch := reShortTermAvg.FindStringSubmatch(match[0]); shortTermAvgMatch != nil {
+			sws.shortTermAvg = shortTermAvgMatch[shortTermAvgIndex]
+		}
+		if longTermAvgMatch := reLongTermAvg.FindStringSubmatch(match[0]); longTermAvgMatch != nil {
+			sws.longTermAvg = longTermAvgMatch[longTermAvgIndex]
+		}
+
+		metricName := match[nameIndex]
+		result[metricName] = sws
+	}
+
+	return result
+}
+
+// stopwatchShowMetricsUpdater updates the metric by obtaining the stopwatch/show
+// metrics for the specified component.
+func stopwatchShowMetricsUpdater(component string) {
+	for {
+		time.Sleep(30 * time.Second)
```

Comment:
I don't think this is optimal here. You should use a ticker instead: https://gobyexample.com/tickers . Tickers take into account the time of execution by the bloc of code below and adapts the succeeding tick to it - as to ensure that you do tick every 30 seconds. Sleep does not, see: https://stackoverflow.com/a/53315789. Normally I would consider this a nit, but since we are exposing metrics to a Prometheus server I think it is important that we consider this, presumably on a cluster under heavy load: exec'ing the ovn-controller / northd binaries might be slower than expected - which might distort our metrics.